internal/preflight/dryrun.go

```diff
@@ -12,25 +15,63 @@ type CreationDryRun struct {
 }
 
 func NewCreationDryRun(client client.Writer) *CreationDryRun {
-	return &CreationDryRun{
-		client: client,
-	}
+	return &CreationDryRun{client: client}
 }
 
-func (p *CreationDryRun) Check(
-	ctx context.Context, _, obj client.Object,
-) (violations []Violation, err error) {
+func (p *CreationDryRun) Check(ctx context.Context, _, obj client.Object) (violations []Violation, err error) {
 	defer addPositionToViolations(ctx, obj, &violations)
 
-	drerr := p.client.Create(ctx, obj.DeepCopyObject().(client.Object), client.DryRunAll)
-	if errors.IsAlreadyExists(drerr) {
+	err = p.client.Create(ctx, obj.DeepCopyObject().(client.Object), client.DryRunAll)
+
+	if k8serrs.IsAlreadyExists(err) {
+		obj := obj.DeepCopyObject().(*unstructured.Unstructured)
+		err = p.client.Patch(ctx, obj, client.MergeFrom(obj), client.DryRunAll)
```

Comment:
Please change to an Server-Side-Apply Patch Type, like in the actual reconciler:
https://github.com/package-operator/package-operator/blob/main/internal/controllers/phase_reconciler.go#L585-L595